pallene/checker.lua

```diff
@@ -243,14 +243,9 @@ function Checker:check_program(prog_ast)
             for _, name in ipairs(top_level_names) do
                 local old_location = names[name]
                 if old_location then
-                    if top_level_node._tag == "ast.Toplevel.Func" or
-                        top_level_node._tag == "ast.Toplevel.Var" then
-                        if not top_level_node.is_local then
-                            scope_error(node_location,
-                                "duplicate export '%s', previous one at line %d",
-                                name, old_location.line)
-                        end
-                    end
+                    scope_error(node_location,
+                        "duplicate '%s', previous one at line %d",
+                        name, old_location.line)
```

Comment:
Let's use the old error message: `duplicate toplevel declaration for '%s'`.

We should also update the comment again to mention that, in addition to avoiding confusion with `export`, this also avoids confusion with the mutually recursive functions.